Modules/CalcDefence-3_0.lua

```diff
@@ -39,6 +39,36 @@ function calcs.armourReduction(armour, raw)
 	return round(calcs.armourReductionF(armour, raw))
 end
 
+-- find second minimum
+function m_second_min(array)
+	local minimum = 2147483648
+	local SecondMinimum = 2147483648
+	for i, val in ipairs(array) do
+		if val < minimum then
+			SecondMinimum = minimum
+			minimum = val
+		elseif val < SecondMinimum then
+			SecondMinimum = val
+		end
+	end
+	return SecondMinimum
+end
+
+-- find second maximum
+function m_second_max(array)
+	local maximum = -2147483648
+	local SecondMaximum = -2147483648
+	for i, val in ipairs(array) do
+		if val > maximum then
+			SecondMaximum = maximum
+			maximum = val
+		elseif val > SecondMaximum then
+			SecondMaximum = val
+		end
+	end
+	return SecondMaximum
+end
+
```

Comment:
Can this be avoided using sort?

Reply:
https://www.lua.org/pil/19.3.html

Reply:
(for small tables should be similar runtime)

Reply:
Removed